schematics/exceptions.py

```diff
@@ -66,55 +129,55 @@ def __init__(self, *args, **kwargs):
                 items = [arg]
         else:
             items = args
-        self.messages = []
+        errors = []
         for item in items:
             if isinstance(item, string_type):
-                self.messages.append(ErrorMessage(item))
+                errors.append(ErrorMessage(item))
             elif isinstance(item, tuple):
-                self.messages.append(ErrorMessage(*item))
+                errors.append(ErrorMessage(*item))
             elif isinstance(item, ErrorMessage):
-                self.messages.append(item)
+                errors.append(item)
             elif isinstance(item, self.__class__):
-                self.messages.extend(item.messages)
+                errors.extend(item.errors)
             else:
                 raise TypeError("'{0}()' object is neither a {1} nor an error message."\
                                 .format(type(item).__name__, type(self).__name__))
-        for message in self.messages:
-            message.type = self.type or type(self)
+        for error in errors:
+            error.type = self.type or type(self)
 
-        super(FieldError, self).__init__(self.messages)
+        super(FieldError, self).__init__(None, errors=errors)
 
     def __eq__(self, other):
         if type(other) is type(self):
-            return other.messages == self.messages
+            return other.errors == self.errors
         elif isinstance(other, list):
-            return other == self.messages
+            return other == self.errors
         return False
 
+    def __hash__(self):
+        return super(FieldError, self).__hash__()
+
     def __contains__(self, value):
-        return value in self.messages
+        return value in self.errors
 
     def __getitem__(self, index):
-        return self.messages[index]
+        return self.errors[index]
 
     def __iter__(self):
-        return iter(self.messages)
+        return iter(self.errors)
 
     def __len__(self):
-        return len(self.messages)
+        return len(self.errors)
 
     def __repr__(self):
-        if len(self.messages) == 1:
-            msg_repr = str(self.messages[0])
+        if len(self.errors) == 1:
+            msg_repr = str(self.errors[0])
         else:
-            msg_repr = '[' + str.join(', ', map(str, self.messages)) + ']'
+            msg_repr = '[' + str.join(', ', map(str, self.errors)) + ']'
         return '%s(%s)' % (self.__class__.__name__, msg_repr)
 
     __str__ = __repr__
```

Comment:
We need to solve the quoting issue and probably return a JSON parseable string:

```python
try:
    Player({'id': 1.2, 'location': [2]})
except DataError as e:
    print(str(e))
```

Now:
```
{'id': u'ConversionError("Value \'1.2\' is not int.")', 'location': u'ConversionError("Input must be a mapping or \'Location\' instance")'}
```

Desired:
```
{"id": "Value '1.2' is not int.", "location": "Input must be a mapping or 'Location' instance"}
```